### PR TITLE
Add prefer_zero_over_explicit_init rule

### PR DESCRIPTION
**Pull request by MortyMerr**
Implements #3190

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/Style/PreferZeroOverExplicitInitRule.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        let secondArg = "(\(["y", "dy", "height"].joined(separator: "|")))"
+        let thirdAndFourthArg = "(\\,\\s*width\(zero)\\,\\s*height\(zero))?"
+        return "\(type)\\(\\s*\(firstArg)\(zero)\\,\\s*\(secondArg)\(zero)\(thirdAndFourthArg)\\)"
+    }
+
+    public static let description = RuleDescription(
+        identifier: "prefer_zero_over_explicit_init",
+        name: "Prefer Zero Over Explicit Init",
+        description: "Prefer `.zero` over excplicit init with zero parameters (e.g. `CGPoint(x: 0, y: 0)`)",
+        kind: .style,
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Style/PreferZeroOverExplicitInitRule.swift:20]: can you update this to be `.idiomatic` and move the rule to the `Idiomatic` folder?

File: Source/SwiftLintFramework/Rules/Style/PreferZeroOverExplicitInitRule.swift [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        let firstArg = "(\(["x", "dx", "width"].joined(separator: "|")))"
+        let secondArg = "(\(["y", "dy", "height"].joined(separator: "|")))"
+        let thirdAndFourthArg = "(\\,\\s*width\(zero)\\,\\s*height\(zero))?"
+        return "\(type)\\(\\s*\(firstArg)\(zero)\\,\\s*\(secondArg)\(zero)\(thirdAndFourthArg)\\)"
+    }
+
+    public static let description = RuleDescription(
+        identifier: "prefer_zero_over_explicit_init",
+        name: "Prefer Zero Over Explicit Init",
+        description: "Prefer `.zero` over excplicit init with zero parameters (e.g. `CGPoint(x: 0, y: 0)`)",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Style/PreferZeroOverExplicitInitRule.swift:19]: typo in `excplicit`

File: Source/SwiftLintFramework/Rules/Style/PreferZeroOverExplicitInitRule.swift [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        kind: .style,
+        minSwiftVersion: .fiveDotOne,
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/Style/PreferZeroOverExplicitInitRule.swift:21]: is there a reason why this is needed?

**Comment by marcelofabri** [comment]
This is awesome, @MortyMerr! 💯

**Comment by MortyMerr** [comment]
@marcelofabri sorry for the late response :) And thank you for fixing issues for me!

**Pull request merged**